agent/agentexec/main_test.go

```diff
@@ -0,0 +1,27 @@
+package agentexec_test
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+)
+
+const TestBin = "/tmp/agent-test"
```

Comment:
This will not work if multiple tests run simultaneously. I would generate a temp dir (manually since you have no testing.T) in `TestMain` and store the path in a global var instead.

And have the `TestMain` try to clean up the directory after `m.Run()` before calling `os.Exit()`